lib/geared_pagination/portions/portion_at_cursor.rb

```diff
@@ -20,7 +20,11 @@ def from(scope)
     end
 
     def next_param(scope)
-      Cursor.encode page_number: page_number + 1, values: from(scope).last&.slice(*attributes) || {}
+      if scope.order_values.none? && scope.limit_value.nil?
```

Comment:
Do we need this check?

Reply:
The scope argument can be a full set of records, or now just a portion and the method supports both.

This check supports the full set of records. Although we now never call this method with a full set of records internally to Geared Pagination, there's a test for it and I wasn't sure about how much of this should be considered a public API?

https://github.com/basecamp/geared_pagination/blob/bfce1030260572d035c3ffd6957c58e2b531b15f/test/portion_at_cursor_test.rb#L55-L68

Reply:
Makes sense thanks! Could extract a method for the check?

Reply:
Pushed https://github.com/basecamp/geared_pagination/pull/55/commits/a3f9008b29a466bae372e6c7f7f688b07ee66873 to avoid the check, by explicitly passing a `portion` in a kwarg without breaking the existing API.

Reply:
Pushed https://github.com/basecamp/geared_pagination/pull/55/commits/22e140bd473949b7920b62eea80a9ca00841ceb4 as per your idea @djmb to memoize instead. Thanks!